### PR TITLE
Add useful web3-related fixtures

### PR DESCRIPTION
**Pull request by pcppcp**

### CODE REVIEW

File: raiden_libs/test/fixtures/contracts.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def contracts_manager():
+    """Reimplement this - fixture should return raiden-lib's contract manager.
+        (return ContractManager(CONTRACTS_SOURCE_DIRS))
+    """
+    raise NotImplementedError(
+        "In order to use contract fixtures please override contracts_manager fixture."""
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/contracts.py]: Use single quotes here. Also I guess you did not want to close it with a triple quote ;)

File: raiden_libs/test/fixtures/contracts.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+from eth_utils import is_address
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def contracts_manager():
+    """Reimplement this - fixture should return raiden-lib's contract manager.
+        (return ContractManager(CONTRACTS_SOURCE_DIRS))
+    """
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/contracts.py]: No newline needed before end of docstring

File: raiden_libs/test/fixtures/contracts.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    raise NotImplementedError(
+        "In order to use contract fixtures please override contracts_manager fixture."""
+    )
+
+
+@pytest.fixture
+def contract_deployer_address(faucet_address) -> str:
+    """Reimplement this - fixture should return an address of the account"""
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/contracts.py]: Wtf? A docstring should only have one pair of `""""`

File: raiden_libs/test/fixtures/contracts.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    """that has enough eth to deploy the contracts."""
+    raise NotImplementedError(
+        "Address of a deployer account must be overriden."""
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/contracts.py]: Same as above. Use single quotes and end it with only one quote.

File: raiden_libs/test/fixtures/contracts.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    )
+
+
+@pytest.fixture
+def deploy_tester_contract(web3,
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/contracts.py]: Please follow the style guide with the arguments here

File: raiden_libs/test/fixtures/contracts.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+                           contracts_manager,
+                           deploy_contract,
+                           contract_deployer_address,
+                           wait_for_transaction,
+                           get_random_address):
+    """Returns a function that can be used to deploy a named contract, """
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/contracts.py]: same problem as above with the docstring

File: raiden_libs/test/fixtures/web3.py [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+from web3 import Web3
+from web3.providers.eth_tester import EthereumTesterProvider
+from web3.contract import Contract
+from web3.providers.rpc import HTTPProvider
+
+from raiden_libs.utils import (
+    address_from_signature,
+    keccak256,
+)
+# import microraiden.utils.contract
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/web3.py]: remove commented out code

File: raiden_libs/test/fixtures/web3.py [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+    return EthereumTester(PyEVMBackend())
+
+
+@pytest.fixture(scope='session')
+def mine_sync_event():
+    return gevent.event.Event()
+
+
+@pytest.fixture(scope='session', autouse=True)
+def disable_requests_loggin():
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/web3.py]: rename `disable_requests_loggin` to `disable_requests_logging`. Add the ending `g`.

File: raiden_libs/test/fixtures/contracts.py [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+
+@pytest.fixture
+def get_contract_abi(contracts_manager):
+    return contracts_manager.get_contract_abi
+
+
+@pytest.fixture
+def standard_token_network_contract(
+    web3,
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/contracts.py]: Follow indentation of the styling guide

File: raiden_libs/test/fixtures/contracts.py [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+            json_contract['bin'],
+            args
+        )
+        return contract
+    return f
+
+
+@pytest.fixture
+def token_network_contract(
+    deploy_tester_contract,
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/contracts.py]: follow indentation of the styling guide

File: raiden_libs/test/fixtures/web3.py [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+        txhash = contract.deploy({'from': deployer_address}, args=args)
+        contract_address = web3.eth.getTransactionReceipt(txhash).contractAddress
+        web3.testing.mine(1)
+
+        return contract(contract_address)
+    return fn
+
+
+@pytest.fixture(scope='session')
+def web3(use_tester: bool,
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/web3.py]: Follow the style guide for function definitions.

File: raiden_libs/test/fixtures/web3.py [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+            filter_ = contract.web3.eth.filter(filter_params)
+            response = contract.web3.eth.getFilterLogs(filter_.filter_id)
+            contract.web3.eth.uninstallFilter(filter_.filter_id)
+            return response
+
+        def patched_wait_for_event(wait: float):
+            web3.testing.mine(1)
+
+        def patched_wait_for_transaction(
+            web3: Web3,
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/web3.py]: Follow the indentation of the style guide please

File: raiden_libs/test/fixtures/web3.py [hunk 13]
```diff
@@ -0,0 +1,208 @@
...
+            timeout: int = DEFAULT_TIMEOUT,
+            polling_interval: int = DEFAULT_RETRY_INTERVAL
+        ):
+            web3.testing.mine(1)
+            tx_receipt = web3.eth.getTransactionReceipt(tx_hash)
+            if tx_receipt is None:
+                raise TimeoutError('Transaction {} was not mined.'.format(tx_hash))
+            return tx_receipt
+
+#        monkeypatch.setattr(microraiden.utils.contract, '_get_logs_raw', patched_get_logs_raw)
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/web3.py]: Remove commented out code.

File: raiden_libs/test/fixtures/web3.py [hunk 14]
```diff
@@ -0,0 +1,208 @@
...
+#            'wait_for_transaction',
+#            patched_wait_for_transaction
+#        )
+
+
+@pytest.fixture(scope='session')
+def wait(web3, kovan_block_time):
+    poll_interval = kovan_block_time / 2
+    gevent.sleep(poll_interval)
+#    return Wait(web3, poll_interval=poll_interval)
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/web3.py]: Does this return nothing? If yes still ... remove commented out code.

File: raiden_libs/test/fixtures/web3.py [hunk 15]
```diff
@@ -0,0 +1,208 @@
...
+    else:
+        yield
+
+
+def token_contract(web3: Web3, token_address: str, token_abi):
+    return web3.eth.contract(abi=token_abi, address=token_address)
+
+
+def token_networks_registry_contract(
+    web3: Web3,
```
Comment (LefterisJP) [inline raiden_libs/test/fixtures/web3.py]: Follow the indentation style guide please

File: raiden_libs/test/fixtures/web3.py [hunk 16]
```diff
@@ -0,0 +1,157 @@
...
+FAUCET_ALLOWANCE = 100 * denoms.ether
+INITIAL_TOKEN_SUPPLY = 200000000000
+WEB3_PROVIDER_DEFAULT = 'http://localhost:8545'
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='session')
+def ethereum_tester():
+    """Return instance of an Ethereum tester"""
```
Comment (palango) [inline raiden_libs/test/fixtures/web3.py]: *Returns an instance...* 

There are other instances of *return* vs. *returns*

File: raiden_libs/test/fixtures/web3.py [hunk 17]
```diff
@@ -0,0 +1,157 @@
...
+    return EthereumTester(PyEVMBackend())
```
Comment (palango) [inline raiden_libs/test/fixtures/web3.py]: Should we make the backend change-able?

Comment (pcppcp) [reply raiden_libs/test/fixtures/web3.py]: unfortunately other backends didn't work for me. I'd leave this as it is now, if someone wants to use different backend he can simply override the fixture.

File: raiden_libs/test/fixtures/web3.py (hunk 7 above)
Comment (LefterisJP) [reply raiden_libs/test/fixtures/web3.py]: The commented out code is still there @pcppcp

File: raiden_libs/test/fixtures/web3.py (hunk 17 above)
Comment (palango) [reply raiden_libs/test/fixtures/web3.py]: Fine for me. Good when py-evm works as we should use that in the long-term.

File: raiden_libs/test/fixtures/web3.py (hunk 7 above)
Comment (palango) [reply raiden_libs/test/fixtures/web3.py]: @pcppcp this is still here ;)

Comment (pcppcp) [reply raiden_libs/test/fixtures/web3.py]: ah damn to hell with that sneaky comment

**Pull request merged**